staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -434,7 +435,7 @@ func withDescription(s spec.Schema, desc string) spec.Schema {
 
 func buildDefinitionsFunc() {
 	namer = openapi.NewDefinitionNamer(runtime.NewScheme())
-	definitions = generatedopenapi.GetOpenAPIDefinitions(func(name string) spec.Ref {
+	definitions = utilopenapi.GetOpenAPIDefinitionsWithoutDisabledFeatures(generatedopenapi.GetOpenAPIDefinitions)(func(name string) spec.Ref {
```

Comment:
apiextensions-apiserver already has place for cleaning up the spec: pkg/controller/openapi/v2.

I wonder, don't we publish the enums for CRDs not already for ages? Wouldn't this PR be a regression?

Reply:
> I wonder, don't we publish the enums for CRDs not already for ages? Wouldn't this PR be a regression?

That's a good question. I'd like to see a test that exercises the behavior of enums in a CRD schema → content of /openapi/v2 at HEAD before the changes in this PR, to make sure we don't regress.

Reply:
The approach is to wrap the generated OpenAPI schemas of built-in types to remove enums (and thus “restore” the definitions). Because after getting the generated built-in defs, this won’t touch anything other than CRD. Also, if it really remove the enums from CRDs, from /openapi/v2 it would have failed the e2e suite.

Reply:
Another note: `pkg/controller/openapi/v2` works with `structuralschema.Structural` but not the `spec.Schema` we want here.